app/coffee/UpdatesManager.coffee

```diff
@@ -188,14 +188,13 @@ module.exports = UpdatesManager =
 		for update in updates
 			earliestUpdate = summarizedUpdates[summarizedUpdates.length - 1]
 			if earliestUpdate and earliestUpdate.meta.start_ts - update.meta.end_ts < @TIME_BETWEEN_DISTINCT_UPDATES
-				if update.meta.user?
-					userExists = false
-					for user in earliestUpdate.meta.users
-						if user.id == update.meta.user.id
-							userExists = true
-							break
-					if !userExists
-						earliestUpdate.meta.users.push update.meta.user
+				userExists = false
+				for user in earliestUpdate.meta.users
```

Comment:
Could you add some comments about what is going on, the logic seems to be getting complicated.
Seems like it all comes down to "add entry to array if not present"!